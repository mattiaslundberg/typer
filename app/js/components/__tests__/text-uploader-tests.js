import React from 'react'
import TextUploader from '../text-uploader'
import sd from 'skin-deep'

describe("TextSelectorTest", () => {
  const render = (onNewCancel=jest.fn()) => {
    return sd.shallowRender(
      <TextUploader onNewCancel={onNewCancel} />
    )
  }

  it('Render input form', () => {
    let tree = render()
    let instance = tree.getMountedInstance()

    expect(tree.getRenderOutput().props.children).toMatchSnapshot()
  })

  // TODO: DOM Tests

  it('Render errors', () => {
    let tree = render()
    let instance = tree.getMountedInstance()
    instance.setState({error: {message: "Something went wrong"}})

    expect(tree.getRenderOutput()).toMatchSnapshot()
  })
})
