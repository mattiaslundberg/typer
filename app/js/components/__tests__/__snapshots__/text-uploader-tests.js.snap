exports[`TextSelectorTest Render errors 1`] = `
<div
  className="error-message">
  Something went wrong
</div>
`;

exports[`TextSelectorTest Render input form 1`] = `
Array [
  <p>
    <label
        htmlFor="name">
        Name
    </label>
    <input
        name="name"
        type="text" />
</p>,
  <p>
    <label
        htmlFor="text">
        Text
    </label>
    <input
        name="text"
        type="text" />
</p>,
  <p>
    <button
        onClick={[Function]}>
        Save
    </button>
    <button
        onClick={undefined}>
        Cancel
    </button>
</p>,
]
`;
